src/main/java/net/minecraftforge/common/crafting/CraftingHelper.java

```diff
@@ -235,4 +253,24 @@ public static ICondition getCondition(JsonObject json)
             throw new JsonSyntaxException("Unknown condition type: " + condition.getID().toString());
         return serializer.getJson(condition);
     }
+
+    @Nullable
+    public static ICondition.IContext buildContext(@Nullable TagManager tagManager) {
+        if (tagManager == null) return null;
+
+        Map<ResourceKey<?>, Map<ResourceLocation, Tag<Holder<?>>>> tagsMap = new IdentityHashMap<>();
+        for (var result : tagManager.getResult())
+        {
+            tagsMap.put(result.key(), Collections.unmodifiableMap((Map) result.tags()));
+        }
+
+        return new ICondition.IContext() {
+            @Override
+            @Nullable
+            public <T> Map<ResourceLocation, Tag<Holder<T>>> getTags(ResourceKey<? extends Registry<T>> key)
```

Comment:
For an alternative to consider, could we make this `Tag<Holder<T> getTags(TagKey<T>)` instead? Its arguably slightly less efficient if you need multiple tags, but makes the interface a lot cleaner.

Reply:
Yes absolutely, implemented!